src/components/craft/inputs/SpaceStyleInput.tsx

```diff
@@ -1,45 +1,152 @@
-import { Form } from 'antd'
-import React from 'react'
-import { defineMessages, useIntl } from 'react-intl'
-import { CSSObject } from 'styled-components'
+import { Form, Input, InputNumber, Radio } from 'antd'
+import React, { useState } from 'react'
+import { useIntl } from 'react-intl'
+import styled, { CSSObject } from 'styled-components'
+import SampleMargin from '../../../images/icon/sample-margin.svg'
+import SamplePadding from '../../../images/icon/sample-padding.svg'
+import { CraftSettingLabel } from '../../../pages/CraftPageAdminPage/CraftSettingsPanel'
 import craftMessages from '../translation'
-import BoxModelInput from './BoxModelInput'
+
+const StyledLabel = styled(CraftSettingLabel)`
+  width: 25%;
+  text-align: center;
+`
+const StyledInputNumber = styled(InputNumber)`
+  width: 25% !important;
+`
+const pxFormatter = (value?: string | number) => {
+  if (value === 'px') {
+    return `0px`
+  }
+  switch (typeof value) {
+    case 'string':
+      return `${value.replace(/[^0-9]/gi, '')}px`
+    case 'number':
+      if (!isNaN(value)) {
+        return `${value}px`
+      } else {
+        return `0px`
+      }
+    default:
+      return `0px`
+  }
+}
 
 export type SpaceStyle = Pick<CSSObject, 'margin' | 'padding'>
-const messages = defineMessages({
-  margin: { id: 'craft.inputs.margin', defaultMessage: '外距' },
-  padding: { id: 'craft.inputs.padding', defaultMessage: '內距' },
-})
+
 const SpaceStyleInput: React.VFC<{
   value?: SpaceStyle
   onChange?: (value: SpaceStyle) => void
 }> = ({ value, onChange }) => {
+  const [spaceTab, setSpaceTab] = useState('padding')
+
+  const paddingArr: string[] | null =
+    typeof value?.padding === 'string' && value.padding.split(' ').length === 4
+      ? value?.padding.split(' ')
+      : ['0px', '0px', '0px', '0px']
+
+  const marginArr: string[] | null =
+    typeof value?.margin === 'string' ? value?.margin.split(' ') : ['0px', '0px', '0px', '0px']
+
+  const onSpacingChange = (adjustedPx: number | string | undefined | null, order: number) => {
+    if (adjustedPx === null || typeof adjustedPx === undefined || adjustedPx === 'px') {
+      adjustedPx = 0
+    }
+    if (spaceTab === 'margin') {
+      marginArr[order] = `${typeof adjustedPx === 'string' ? adjustedPx.replace(/[^0-9]/gi, '') : adjustedPx}px`
+      let margin = marginArr.join(' ')
+      onChange?.({ ...value, margin })
+    } else if (spaceTab === 'padding') {
+      paddingArr[order] = `${typeof adjustedPx === 'string' ? adjustedPx.replace(/[^0-9]/gi, '') : adjustedPx}px`
+      let padding = paddingArr.join(' ')
+      onChange?.({ ...value, padding })
+    }
+  }
+
+  const handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>, order: number) => {
+    onSpacingChange(e.currentTarget.value, order)
+  }
+
   const { formatMessage } = useIntl()
```

Comment:
NITS:  can move up to line 41

Reply:
@Ponchimeow  I had moved. You can see commit below this comment